fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4

```diff
@@ -1234,16 +1234,16 @@ grantPrivilegeStatement
     : GRANT IMPERSONATE ON user TO grantRevokeClause                                                     #grantImpersonateBrief
     | GRANT privilegeActionList ON tableDbPrivilegeObjectNameList TO grantRevokeClause                   #grantTablePrivBrief
     | GRANT privilegeActionList ON privilegeType (privilegeObjectNameList)? TO grantRevokeClause         #grantPrivWithType
-    | GRANT privilegeActionList ON privilegeType qualifiedName '(' typeList ')' TO grantRevokeClause     #grantPrivWithFunc
-    | GRANT privilegeActionList ON ALL privilegeType (IN ALL privilegeType)* (IN privilegeType identifierOrString)? TO grantRevokeClause   #grantOnAll
+    | GRANT privilegeActionList ON GLOBAL? privilegeType qualifiedName '(' typeList ')' TO grantRevokeClause     #grantPrivWithFunc
```

Comment:
this is bit of tricky, can we use a separate clause?

Reply:
OK. I'll try my best.

Reply:
I think it's okay to use first clause, because actually this clause is right for function. No need to add a extra clause.

```
    | GRANT privilegeActionList ON GLOBAL? privilegeType qualifiedName '(' typeList ')' TO grantRevokeClause     #grantPrivWithFunc
```

But for `ALL GLOBAL FUNCTIONS`,   I think it's okay to add a separated clause like

```
    | GRANT privilegeActionList ON ALL GLOBAL FUNCTIONS TO grantRevokeClause   #grantOnAllGlobalFunctions

```

What do you think?

Reply:
i agree

Reply:
@nshangyiming Updated.